metadata-ingestion/sink_overview.md

```diff
@@ -0,0 +1,28 @@
+# Sinks
+
+Sinks are **destinations for metadata**.
```

Comment:
it feels like we might need a diagram that shows how sources, sinks, recipes, and datahub all interact with each other

we have this diagram https://datahubproject.io/docs/metadata-ingestion/developing/#architecture which is kinda outdated and a bit too technical (don't need the MetadataChangeEvent in there), but a variant of this diagram would probably do wonders for understandability

we can put that diagram at the top of the ingestion/source/sink pages